types/react/index.d.ts

```diff
@@ -1118,7 +1118,7 @@ declare namespace React {
         which: number;
     }
 
-    interface MouseEvent<T = Element, E = NativeMouseEvent> extends SyntheticEvent<T, E> {
+    interface MouseEvent<T = Element, E = NativeMouseEvent> extends UIEvent<T, E> {
```

Comment:
Reacts synthetic events and native DOM events are two different things. Please include documentation that shows that properties of UIEvent are part of reacts synthetic events. The native mouse events should already inherit from UIEvent.

Reply:
https://github.com/facebook/react/blob/master/packages/react-dom/src/events/SyntheticMouseEvent.js#L21

Looks like it does extend the (Synthetic)UIEvent.

Reply:
`.extend` is itself https://github.com/facebook/react/blob/master/packages/legacy-events/SyntheticEvent.js#L227-L249 which is an ES5 implementation of `class` inheritance.

Reply:
(Synthetic)`TouchEvent` and (Synthetic)`KeyboardEvent` also do `.extend` from (Synthetic)`UIEvent` so it looks like this PR is in the right direction, just with a smaller scope than it could be.

Reply:
Thanks @eps1lon @Jessidhia for looking at the pr.

@Jessidhia you beat me to it! I found the link in the source code last night but didn't have the chance to comment on the pr until now. I did try to look through the documentation otherwise and couldn't find a reference to the direct link there.

Other than the build failure, I'm unclear if I need to make further changes for this pr to be considered ready to merge?

On the build failure front, is there a cut down version of the tests I can run to replicate the problem? I can't see any obvious switches/options to pass to the test runner to do so. On the face of the error, it would appear unrelated to my change, but the feedback loop is very long (test section in Azure is about 26 minutes...!).

Reply:
> Other than the build failure, I'm unclear if I need to make further changes for this pr to be considered ready to merge?

I think you're extending the native UIEvent. You should extend the SyntheticUIEvent.

Reply:
No, it is the locally-declared `UIEvent`. The events here are declared without a prefix (it's `React.UIEvent`, `React.MouseEvent`, etc, but since we're already inside the `React` namespace we don't use the `React.` prefix)

The native events are shadowed and aliased with a `Native` prefix, which is why `E = NativeMouseEvent`.

Reply:
Then `SyntheticMouseEvent` shouldn't have been there in the first place anymore?